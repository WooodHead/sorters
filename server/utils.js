function prepare(o) {
    if (o && o._id) {
        o._id = o._id.toString()
    }
    return o
}

async function findOne(Collection, ...args) {
    return prepare(await Collection.findOne(...args))
}

async function find(Collection, ...args) {
    return (await Collection.find(...args).toArray()).map(prepare)
}

module.exports = {
    prepare,
    findOne,
    find,
}