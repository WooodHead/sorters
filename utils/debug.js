
export function log(arg) {
    console.log(arg)
    return arg
}
