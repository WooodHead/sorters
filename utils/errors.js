export const errorMessage = (e) => {
    if (e.graphQLErrors && e.graphQLErrors.length > 0) {
        return e.graphQLErrors[0].message
    } else {
        return e.message
    }
}