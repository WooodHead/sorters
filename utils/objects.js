
function toObject(arr) {
    const res = {}
    for (const [key, value] of arr) {
        res[key] = value
    }
    return res
}

function objectMap(object, keyMap = key => key, valueMap = value => value) {
    return Object.keys(object).reduce((o, key) => {
        o[keyMap(key)] = valueMap(object[key])
        return o
    }, {})
}

module.exports = {
    toObject,
    objectMap
}
