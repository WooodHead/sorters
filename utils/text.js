export const replaceAll = (string, search, replacement) => {
  if (typeof string === "string") {
    return string.replace(new RegExp(search, 'g'), replacement)
  } else {
    return string
  }
}

export const teaserAndInfo = (content, min, opt, max) => {
	if (content) {
		if (content.length <= max) {
			return {
				cut: false,
				teaser: content
			};
		}
		const words = content.split(" ");
		let teaser = "";
		for (let i=0; i<words.length; i++) {
			const candidate = words[i];
			let wouldbelength = teaser.length + 1 + candidate.length + 3;
			if (wouldbelength >= opt) {
				// We're done
				if (wouldbelength > max) {
					if (teaser.length + 3 < min) {
						// cut the word
						return {
							teaser: teaser + " " + candidate.substring(0, opt - teaser.length - 1) + "...",
							cut: true,
						};
					} else {
						return {
							teaser: teaser + "...",
							cut: true,
						};
					}
				} else {
					return {
						teaser: teaser + " " + candidate + "...",
						cut: true,
					};
				}
			}
			teaser += " " + candidate;
		}
		return {
			teaser: teaser + "...",
			cut: true,
		};
	} else {
		return {
			teaser: "",
			cut: false,
		};
	}
};
export const teaser = (content, min, opt, max) => {
	const teaser = teaserAndInfo(content, min, opt, max);
	return teaser.teaser;
}
