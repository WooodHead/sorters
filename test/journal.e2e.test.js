import cheerio from 'cheerio'
import {pretty} from './utils'
import {ObjectID} from 'mongodb'
import {setup, teardown} from './setup'
import {generateAndLogUser, getGoalIds} from './fixtures'
import {request} from './utils'

let db
let browser

beforeAll(async () => {
    const res = await setup()
    db = res.db
    browser = res.browser
})

afterEach(async () => {
    await db.dropDatabase()    
})

afterAll(async () => {
    await teardown({db, browser})
})

describe('journal', () => {
    it('displays', async () => {
        const browserPage = await browser.createPage()
        
        await generateAndLogUser(browserPage)

        const goalIds = await getGoalIds(browserPage)

        const res = await request(browserPage, `http://localhost:3000/graphql`, 'POST', {
            query: `
                mutation($entry: NewEntryInput!) {
                    createEntry(entry: $entry) {
                        _id
                    }
                }
            `,
            variables: {
                entry: {
                    title: 'Hello World',
                    url: 'http://www.example.com',
                    description: 'Lorem ipsum',
                    goalIds,
                },
            },
        })

        const status = await browserPage.open(`http://localhost:3000/account/journal`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })
})