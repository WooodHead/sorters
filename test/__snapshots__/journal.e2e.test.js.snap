// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`journal displays 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <div style=\\"padding-top:50px;padding-bottom:20px\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\" href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li>
                <a href=\\"/\\">Home</a>
              </li>
              <li>
                <a href=\\"/about\\">About</a>
              </li>
              <li>
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"dropdownundefined\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\">Hello, test <span class=\\"caret\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <div>
        <h1>Account</h1>
        <ul class=\\"nav nav-tabs\\">
          <li role=\\"presentation\\">
            <a href=\\"/account\\">&#x2302; Dashboard</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/profile\\">&#x1F464; Profile</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/messages\\">&#x1F5E9; Messages</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/goals\\">&#x25CE; Goals</a>
          </li>
          <li role=\\"presentation\\" class=\\"active\\">
            <a href=\\"/account/journal\\">&#x270E; Journal</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/topics\\">&#x1F4A1; Topics</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/essays\\">&#x270E; Essays</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/speeches\\">&#x1F444; Speeches</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/conversations\\">&#x1F5E9; Conversations</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/reads\\">&#x1F4D6; Reading List</a>
          </li>
          <li role=\\"presentation\\">
            <a href=\\"/account/account\\">&#x2699; Account</a>
          </li>
        </ul>
      </div>
      <h2>Journal</h2>
      <p>Here you can write regular updates about your
        <a href=\\"/account/goals\\">goals</a>.</p>
      <div>
        <p>Your journal can be found at
          <a href=\\"/u/test/journal\\">/u/ test /journal</a>.</p>
        <button class=\\"btn btn-primary\\">New Journal Entry</button>
        <div>
          <h3>Entries</h3>
          <div style=\\"margin-bottom:1.5rem\\">
            <div style=\\"margin-top:1.5rem;margin-bottom:1.5rem\\"><span style=\\"display:block;float:right\\"><span><a style=\\"cursor:pointer;font-size:140%\\">&#x1F5D1;</a></span>
              <a style=\\"cursor:pointer;font-size:140%\\">&#x270E;</a>
              </span>
              <h4>
                <a href=\\"http://www.example.com\\" target=\\"_blank\\">Hello World</a>
              </h4>
              <div class=\\"markdown\\">
                <p>Lorem ipsum</p>
              </div>
              <a href=\\"/entry/<id>\\">Comments</a>
              <hr>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>. Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>.</p>
      </div>
    </footer>
  </div>
</div>"
`;
